posts/tree-cell-typing/index.qmd

```diff
@@ -0,0 +1,241 @@
+```
+
+As two examples from this heatmap, we can see that T cell subtypes as well as NK cells have high similarity and so should be grouped together. Additionally, endothelial and fibroblast cell types are highly similar, supporting our grouping of them into a 'structural' major cell type.
+
+Once we have this cell type tree, we can use `InSituTree` to run supervised cell typing with a nested approach.
+```{r runInSituTree, message=FALSE, eval = FALSE}
+# Load basic data consisting of:
+#   - counts: cell x gene count matrix
+#   - negmeans: vector of mean negative probe counts per cell
+
+load("mini_nsclc")
+
+# Run InSituTree with basic options
+res <- runInSituTree(
+  x = mini_nsclc$counts,
+  neg = rowMeans(mini_nsclc$neg),
+  full_profiles = ioprofiles,
+  cth = celltype_hierarchy_list,
+  cohort = NULL,
+  excluded_genes = NULL,
+  quantile_absolute_expression_difference_param = 0.5, # Include genes in the top 0.5 fraction in absolute expression differences
+  quantile_percent_expression_difference_param = 0.5, # Include genes in the top 0.5 fraction in percentage expression differences
+  return_summary_annotation = TRUE
+)
+
+# Access the final summarized cell typing results
+head(res$summaryAnnotation, 10)
+
+```
+
+```{r echo = FALSE, warning = FALSE, eval = TRUE}
+
+res <- readRDS("./assets/small_results.rds")
+
+head(res, 10)
+```
+
+# Additional recommendations {#sec-additional-recommendations}
+The structure of the cell type hierarchy list is a key determinant of how well this approach works. Major cell types that are unnecessarily broad may lead to poor cell typing if the average gene expression is too distinct from the expression of individual cells. As an example, the cell type lists available at our [CosMx Cell Profile library](https://github.com/Nanostring-Biostats/CosMx-Cell-Profiles/tree/main){target="_blank"} often contain an initial grouping of cell types into 'Tissue' or 'Immune'; we would recommend removing that large group and providing the next level down as top annotations. `InSituTree` does run quickly, so in cases of uncertainty we would recommend running the algorithm with multiple cell type hierarchies and choosing the one that gives the best cell typing. Evaluate the results as cell typing is typically evaluated: spatial distribution of cell types, top markers for each cell type, and clustering of similar cell types in UMAP space.
+
+The approach above only works for fully supervised cell typing. In datasets where you have novel cell types not captured in your reference profile, one option is first to run semi-supervised `InSituType` on your whole dataset. Second, you could either create a reference profile from your CosMx data and run `InSituTree` using these new cell type definitions. Alternatively, you could select out only cells of interest from your full dataset cell typing annotations, for example only the immune cells, and run targeted subclustering with `InSituTree` on those cells only.
```

Comment:
The description for alternative option is unclear. Are these 2 meant to address the same issue? or just different use cases  of `InSituTree`? How about this?


> `InSituTree` is designed to work for fully supervised cell typing only. In datasets where you have novel cell types not captured in your reference profile, one option is  to first run semi-supervised InSituType on your whole dataset to separate between novel cell types from the rest. And then, one can create a reference profile from that semi-supervised cell typing restuls of the CosMx data and further redo cell typing via `InSituTree` using the new reference profiles and a `cell_type_hierachy` list. 
> Another helpful use case for `InSituTree` is to do fine cell typing between closely related subtypes. For a dataset that has been through one iteration of cell typing, one could select out only cells of interest from your full dataset cell typing annotations, for example only the immune cells, and run targeted subclustering with `InSituTree` on those cells only.

Reply:
Agree that this is clearer. I've updated the post.